src/test/scala/org/broadinstitute/dsde/firecloud/service/AgoraEntityServiceSpec.scala

```diff
@@ -0,0 +1,174 @@
+package org.broadinstitute.dsde.firecloud.service
+
+import akka.testkit.TestActorRef
+import org.broadinstitute.dsde.firecloud.FireCloudExceptionWithErrorReport
+import org.broadinstitute.dsde.firecloud.dataaccess.{AgoraException, MockAgoraDAO, StatefulMockAgoraDAO}
+import org.broadinstitute.dsde.firecloud.model.MethodRepository.{EditMethodRequest, EditMethodResponse, MethodId}
+import org.broadinstitute.dsde.firecloud.model._
+import org.broadinstitute.dsde.firecloud.service.PerRequest.RequestComplete
+import org.broadinstitute.dsde.rawls.model.ErrorReport
+import org.scalatest.BeforeAndAfterEach
+import spray.http.StatusCode
+import spray.http.StatusCodes._
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class AgoraEntityServiceSpec extends BaseServiceSpec with BeforeAndAfterEach {
+
+  val dur = Duration(2, MINUTES)
+
+  var dao: StatefulMockAgoraDAO = _
+  var aes: AgoraEntityService = _
+
+  val validPayload = "task wc {File in_file command { cat ${in_file} | wc -l } output { Int count = read_int(stdout()) }} workflow www {call wc}"
```

Comment:
in any test that actually talks to agora - including when we as humans are testing via swagger - beware the payload. Agora validates the payload when creating a method, and an invalid payload will result in an error.